### ISSUE TITLE
[BUG]小数据集用小模型训练mAP为0，用大模型训练mAP95??

### ISSUE THREAD

**Issue by TinyQi** [PaddlePaddle/PaddleDetection#5379]
**PaddleDetection team appreciate any suggestion or problem you delivered~**

## Checklist:

1. 查找历史相关issue寻求解答/I have searched related issues but cannot get the expected help.
2. 翻阅[FAQ](https://paddledetection.readthedocs.io/FAQ.html) /I have read the [FAQ documentation](https://paddledetection.readthedocs.io/FAQ.html) but cannot get the expected help.
3. 确认bug是否在新版本里还未修复/The bug has not been fixed in the latest version.

## 描述问题/Describe the bug
我有一批总共500张数据的小数据集，是不同样式的一块文字区域，
1. 然后我一开始是想用小模型训练（有速度要求），所以先尝试了picodet模型，训练结束后，mAP为0.
2. 我尝试用稍大一点的模型，ppyolo_mbv3_large_coco，仍然为0
3. 最终我尝试了backbone更大的模型，yolov3_darknet53_270e_voc，训练最终收敛，mAP为95

## 复现/Reproduction

1. 您使用的命令是？/What command or script did you run?
小模型的命令：python -m paddle.distributed.launch --gpus 6,7  tools/train.py -c /share/disk2/data/02.detection/04.keyfream_test/ppyolo_large_configs/ppyolo_mbv3_large_coco.yml  --eval --use_vdl=true  2>&1 |tee /share/disk2/data/02.detection/04.keyfream_test/result_weights/ppyolo_mbv3_large_coco/train.lo

大模型的命令：python -m paddle.distributed.launch --gpus 1,2,3,5,6,7  tools/train.py -c /share/disk1/training_tools/PaddleDetection-release-2.2/configs/yolov3/yolov3_darknet53_270e_voc.yml   --eval --use_vdl=true 2>&1 |tee train.log

```none
请填写命令/A placeholder for the command.
```
2. 您是否更改过代码或配置文件？您是否理解您所更改的内容？还请您提供所更改的部分代码。/Did you make any modifications on the code or config? Did you understand what you have modified? Please provide the codes that you modified.
1. 我除了更改了数据集以及一些存储区域之类的参数以外，就只调过学习率，而且学习率是按照多卡就大一点，卡少的话就小一点的原则去修改的
2. 用小模型的时候，我尝试过聚类anchor再训练，小模型仍然mAP为0

5. 您使用的数据集是？/What dataset did you use?

6. 请提供您出现的报错信息及相关log。/Please provide the error messages or relevant log information.

## 环境/Environment
1. 请提供您使用的Paddle和PaddleDetection的版本号/Please provide the version of Paddle and PaddleDetection you use：
PaddleDetection为2.3，PaddlePaddle-gpu为2.2

**Comment by TinyQi** [comment]
小模型的训练过程的日志,另外一个小模型也是如此
[03/16 20:33:16] ppdet.engine INFO: Epoch: [218] [0/4] learning_rate: 0.000436 loss_xy: 0.953550 loss_wh: 1.513660 loss_iou: 3.944998 loss_obj: 6.209154 loss_cls: 1.874908 loss: 14.496269 eta: 3:01:06 batch_cost: 11.7773 data_cost: 10.3369 ips: 1.3585 images/s
[03/16 20:33:18] ppdet.engine INFO: Epoch: [218] [2/4] learning_rate: 0.000437 loss_xy: 1.010105 loss_wh: 1.430049 loss_iou: 4.098546 loss_obj: 6.106553 loss_cls: 1.691010 loss: 14.336263 eta: 3:00:31 batch_cost: 0.5324 data_cost: 0.0002 ips: 30.0499 images/s
[03/16 20:33:19] ppdet.utils.checkpoint INFO: Save checkpoint: /share/disk2/data/02.detection/04.keyfream_test/result_weights/ppyolo_tiny_650e_coco
[03/16 20:33:19] ppdet.engine INFO: Eval iter: 0
[03/16 20:33:20] ppdet.metrics.metrics INFO: Accumulating evaluatation results...
[03/16 20:33:20] ppdet.metrics.metrics INFO: mAP(0.50, 11point) = 0.00%
[03/16 20:33:20] ppdet.engine INFO: Total sample number: 47, averge FPS: 62.935435817966
[03/16 20:33:20] ppdet.engine INFO: Best test bbox ap is 0.002.
[03/16 20:33:41] ppdet.engine INFO: Epoch: [219] [0/4] learning_rate: 0.000438 loss_xy: 0.979578 loss_wh: 1.090656 loss_iou: 3.741701 loss_obj: 8.227566 loss_cls: 2.036458 loss: 16.075958 eta: 3:00:36 batch_cost: 10.8029 data_cost: 9.0771 ips: 1.4811 images/s
[03/16 20:33:42] ppdet.engine INFO: Epoch: [219] [2/4] learning_rate: 0.000439 loss_xy: 0.929049 loss_wh: 1.366353 loss_iou: 3.835424 loss_obj: 6.592034 loss_cls: 1.882670 loss: 14.605530 eta: 3:00:01 batch_cost: 0.5268 data_cost: 0.2569 ips: 30.3726 images/s
[03/16 20:33:45] ppdet.utils.checkpoint INFO: Save checkpoint: /share/disk2/data/02.detection/04.keyfream_test/result_weights/ppyolo_tiny_650e_coco
[03/16 20:33:45] ppdet.engine INFO: Eval iter: 0
[03/16 20:33:46] ppdet.metrics.metrics INFO: Accumulating evaluatation results...
[03/16 20:33:46] ppdet.metrics.metrics INFO: mAP(0.50, 11point) = 0.00%
[03/16 20:33:46] ppdet.engine INFO: Total sample number: 47, averge FPS: 53.578427764680555
[03/16 20:33:46] ppdet.engine INFO: Best test bbox ap is 0.002.
[03/16 20:34:13] ppdet.engine INFO: Epoch: [220] [0/4] learning_rate: 0.000440 loss_xy: 0.852978 loss_wh: 1.218274 loss_iou: 3.501668 loss_obj: 7.158501 loss_cls: 1.944256 loss: 14.675676 eta: 3:00:21 batch_cost: 14.4858 data_cost: 10.1707 ips: 1.1045 images/s
[03/16 20:34:14] ppdet.engine INFO: Epoch: [220] [2/4] learning_rate: 0.000441 loss_xy: 0.922667 loss_wh: 1.282585 loss_iou: 3.700190 loss_obj: 7.526640 loss_cls: 1.929610 loss: 15.361692 eta: 2:59:45 batch_cost: 0.3110 data_cost: 0.0010 ips: 51.4520 images/s
[03/16 20:34:15] ppdet.utils.checkpoint INFO: Save checkpoint: /share/disk2/data/02.detection/04.keyfream_test/result_weights/ppyolo_tiny_650e_coco
[03/16 20:34:15] ppdet.engine INFO: Eval iter: 0
[03/16 20:34:16] ppdet.metrics.metrics INFO: Accumulating evaluatation results...
[03/16 20:34:16] ppdet.metrics.metrics INFO: mAP(0.50, 11point) = 0.00%
[03/16 20:34:16] ppdet.engine INFO: Total sample number: 47, averge FPS: 55.96307666452238
[03/16 20:34:16] ppdet.engine INFO: Best test bbox ap is 0.002.

**Comment by TinyQi** [comment]
大模型的训练过程日志
[03/16 23:24:02] ppdet.engine INFO: Epoch: [255] [0/7] learning_rate: 0.000446 loss_xy: 2.359886 loss_wh: 0.723247 loss_obj: 2.183158 loss_cls: 0.242293 loss: 5.729270 eta: 0:09:18 batch_cost: 4.4531 data_cost: 1.7926 ips: 1.7965 images/s
[03/16 23:24:37] ppdet.engine INFO: Epoch: [256] [0/7] learning_rate: 0.000448 loss_xy: 2.384919 loss_wh: 0.805964 loss_obj: 2.310451 loss_cls: 0.240760 loss: 5.934967 eta: 0:08:40 batch_cost: 4.6240 data_cost: 1.5343 ips: 1.7301 images/s
[03/16 23:25:13] ppdet.engine INFO: Epoch: [257] [0/7] learning_rate: 0.000450 loss_xy: 2.371260 loss_wh: 0.825142 loss_obj: 2.180368 loss_cls: 0.220508 loss: 5.638921 eta: 0:08:03 batch_cost: 4.6415 data_cost: 1.3064 ips: 1.7236 images/s
[03/16 23:25:43] ppdet.engine INFO: Epoch: [258] [0/7] learning_rate: 0.000452 loss_xy: 2.352174 loss_wh: 0.764185 loss_obj: 2.303427 loss_cls: 0.254104 loss: 5.460910 eta: 0:07:26 batch_cost: 4.9588 data_cost: 2.1220 ips: 1.6133 images/s
[03/16 23:26:15] ppdet.engine INFO: Epoch: [259] [0/7] learning_rate: 0.000453 loss_xy: 2.317121 loss_wh: 0.736731 loss_obj: 2.070762 loss_cls: 0.202690 loss: 5.156671 eta: 0:06:48 batch_cost: 4.7971 data_cost: 1.9164 ips: 1.6677 images/s
[03/16 23:26:43] ppdet.utils.checkpoint INFO: Save checkpoint: output/yolov3_darknet53_270e_voc
[03/16 23:26:43] ppdet.engine INFO: Eval iter: 0
[03/16 23:26:46] ppdet.metrics.metrics INFO: Accumulating evaluatation results...
[03/16 23:26:46] ppdet.metrics.metrics INFO: mAP(0.50, 11point) = 95.64%
[03/16 23:26:46] ppdet.engine INFO: Total sample number: 47, averge FPS: 14.914739492543095
[03/16 23:26:46] ppdet.engine INFO: Best test bbox ap is 0.956.
[03/16 23:26:52] ppdet.utils.checkpoint INFO: Save checkpoint: output/yolov3_darknet53_270e_voc
[03/16 23:26:53] ppdet.engine INFO: Epoch: [260] [0/7] learning_rate: 0.000455 loss_xy: 2.386294 loss_wh: 0.724145 loss_obj: 2.175719 loss_cls: 0.201713 loss: 5.361660 eta: 0:06:10 batch_cost: 4.2010 data_cost: 1.6835 ips: 1.9043 images/s
[03/16 23:27:21] ppdet.engine INFO: Epoch: [261] [0/7] learning_rate: 0.000457 loss_xy: 2.331084 loss_wh: 0.830063 loss_obj: 2.262330 loss_cls: 0.187270 loss: 5.629555 eta: 0:05:33 batch_cost: 4.0738 data_cost: 1.2189 ips: 1.9638 images/s
[03/16 23:27:52] ppdet.engine INFO: Epoch: [262] [0/7] learning_rate: 0.000459 loss_xy: 2.328418 loss_wh: 0.784579 loss_obj: 2.415187 loss_cls: 0.215903 loss: 5.827135 eta: 0:04:56 batch_cost: 4.0526 data_cost: 1.5193 ips: 1.9740 images/s
[03/16 23:28:23] ppdet.engine INFO: Epoch: [263] [0/7] learning_rate: 0.000460 loss_xy: 2.409398 loss_wh: 0.862292 loss_obj: 2.415187 loss_cls: 0.190137 loss: 5.905525 eta: 0:04:19 batch_cost: 4.4413 data_cost: 2.1187 ips: 1.8013 images/s
[03/16 23:28:54] ppdet.engine INFO: Epoch: [264] [0/7] learning_rate: 0.000462 loss_xy: 2.383475 loss_wh: 0.834288 loss_obj: 2.216724 loss_cls: 0.179196 loss: 5.853278 eta: 0:03:41 batch_cost: 4.5613 data_cost: 2.4922 ips: 1.7539 images/s
[03/16 23:29:19] ppdet.utils.checkpoint INFO: Save checkpoint: output/yolov3_darknet53_270e_voc
[03/16 23:29:19] ppdet.engine INFO: Eval iter: 0
[03/16 23:29:23] ppdet.metrics.metrics INFO: Accumulating evaluatation results...
[03/16 23:29:23] ppdet.metrics.metrics INFO: mAP(0.50, 11point) = 81.60%
[03/16 23:29:23] ppdet.engine INFO: Total sample number: 47, averge FPS: 15.077072675725246
[03/16 23:29:23] ppdet.engine INFO: Best test bbox ap is 0.956.
[03/16 23:29:25] ppdet.engine INFO: Epoch: [265] [0/7] learning_rate: 0.000464 loss_xy: 2.321176 loss_wh: 0.917418 loss_obj: 2.374473 loss_cls: 0.182154 loss: 5.902145 eta: 0:03:04 batch_cost: 4.1002 data_cost: 2.2223 ips: 1.9511 images/s
[03/16 23:29:55] ppdet.engine INFO: Epoch: [266] [0/7] learning_rate: 0.000466 loss_xy: 2.319195 loss_wh: 0.924197 loss_obj: 2.356995 loss_cls: 0.193047 loss: 5.902145 eta: 0:02:27 batch_cost: 4.0255 data_cost: 1.9765 ips: 1.9873 images/s
[03/16 23:30:28] ppdet.engine INFO: Epoch: [267] [0/7] learning_rate: 0.000467 loss_xy: 2.301609 loss_wh: 0.861972 loss_obj: 2.347691 loss_cls: 0.197542 loss: 5.752944 eta: 0:01:50 batch_cost: 4.1617 data_cost: 2.0719 ips: 1.9223 images/s
[03/16 23:31:01] ppdet.engine INFO: Epoch: [268] [0/7] learning_rate: 0.000469 loss_xy: 2.351303 loss_wh: 0.758163 loss_obj: 1.921118 loss_cls: 0.223447 loss: 5.272093 eta: 0:01:13 batch_cost: 4.7199 data_cost: 2.1624 ips: 1.6950 images/s
INFO 2022-03-16 23:32:08,030 launch.py:285] Local processes completed.

**Comment by qingqing01** [comment]
您的数据集格式是VOC，注意选用VOC格式的配置。COCO和VOC数据格式不同，配置不同。

参考 https://github.com/PaddlePaddle/PaddleDetection/blob/release/2.3/configs/picodet/picodet_s_320_voc.yml

**Comment by TinyQi** [comment]
> 您的数据集格式是VOC，注意选用VOC格式的配置。COCO和VOC数据格式不同，配置不同。
> 
> 参考 https://github.com/PaddlePaddle/PaddleDetection/blob/release/2.3/configs/picodet/picodet_s_320_voc.yml

您好，我确实是使用的voc，然后我把关键配置文件中加载的dataset的配置文件改成voc的配置文件了，不然也跑不起来吧

**Comment by TinyQi** [comment]
> 您的数据集格式是VOC，注意选用VOC格式的配置。COCO和VOC数据格式不同，配置不同。
> 
> 参考 https://github.com/PaddlePaddle/PaddleDetection/blob/release/2.3/configs/picodet/picodet_s_320_voc.yml

您好，我刚刚测试了一下，我发现问题确实出在我用了coco的主配置文件，而且关键参数就是
`EvalReader:
  collate_batch: false`

这个参数是干啥的呢？我加上之后，mAP就有结果了

**Comment by yghstill** [comment]
@TinyQi collate_batch表示是否组batch，voc数据集eval时不组batch，此问题已经在最新代码中修复适配，之后就不用加这个collate_batch: false的字段了。
